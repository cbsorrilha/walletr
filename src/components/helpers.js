import { css } from 'styled-components';

export const media = {
  desktop: 992,
  tablet: 768,
  mobile: 576,
};

export const margin = ({ margin }) => {
  if (!margin) {
    return '';
  }
  if (typeof margin == 'string') {
    return css`
      margin: ${margin};
    `;
  }

  return css`
    ${margin.top ? `margin-top: ${margin.top}` : `margin-top: 0`};
    ${margin.right ? `margin-right: ${margin.right}` : `margin-right: 0`};
    ${margin.bottom ? `margin-bottom: ${margin.bottom}` : `margin-bottom: 0`};
    ${margin.left ? `margin-left: ${margin.left}` : `margin-left: 0`};
  `;
};

export const padding = ({ padding }) => {
  if (!padding) {
    return '';
  }
  if (typeof padding == 'string') {
    return css`
      padding: ${padding};
    `;
  }

  return css`
    ${padding.top ? `padding-top: ${padding.top}` : `padding-top: 0`};
    ${padding.right ? `padding-right: ${padding.right}` : `padding-right: 0`};
    ${padding.bottom ? `padding-bottom: ${padding.bottom}` : `padding-bottom: 0`};
    ${padding.left ? `padding-left: ${padding.left}` : `padding-left: 0`};
  `;
};

export const border = ({ border }) => {
  if (!border) {
    return '';
  }
  if (typeof border == 'string') {
    return css`
      border: ${border};
    `;
  }

  return css`
    ${border.top ? `border-top: ${border.top}` : `border-top: 0`};
    ${border.right ? `border-right: ${border.right}` : `border-right: 0`};
    ${border.bottom ? `border-bottom: ${border.bottom}` : `border-bottom: 0`};
    ${border.left ? `border-left: ${border.left}` : `border-left: 0`};
    ${border.radius ? `border-radius: ${border.radius}` : ``};
  `;
};

export const textStyles = ({ align = 'left', color = '#333', weight = 'normal', bold }) => css`
  ${!align ? '' : `text-align: ${align}`};
  ${!color ? '' : `color: ${color}`};
  ${!weight ? '' : `font-weight: ${weight}`};
  ${!bold ? '' : `font-weight: 700`};
`;
