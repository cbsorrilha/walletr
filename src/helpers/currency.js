export const converterFromPrice = price => [
  value => parseFloat(value) * parseFloat(price),
  value => parseFloat(value) / parseFloat(price),
];
// TODO
export function BTCtoBTA(value, btcPrice, btaPrice) {}
// TODO
export function BTAtoBTC(value, btcPrice, btaPrice) {}
