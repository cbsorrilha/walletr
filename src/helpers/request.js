import Axios from 'axios';
import { API_URL } from '../constants';

export const request = (endpoint, options = { method: 'get' }) => {
  return Axios({
    url: `${API_URL}/${endpoint}`,
    ...options,
  });
};
