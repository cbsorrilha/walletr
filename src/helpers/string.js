export function formatMoneyString(amount) {
  return `R$ ${parseFloat(amount).toFixed(2)}`;
}
