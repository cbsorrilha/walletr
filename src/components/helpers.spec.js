import { media, margin, border, padding } from './helpers';

describe("Test the component helpers", () => {
  describe("Test  the media helper", () => {
    it('should match the snapshots', () => {
      expect(media).toMatchSnapshot();
    });
  });

  describe("Test the margin styled-components helper", () => {
    it('should return nothing if no margin is passed', () => {
      const noMargin = margin({});
      expect(noMargin).toBe('');
    });

    it('should return a string containing the declared string margin', () => {
      const stringMargin = margin({ margin: "10px" });
      expect(stringMargin.join("")).toEqual(expect.stringContaining('margin: 10px;'));
    });

    it('should return a string containing the declared object margin', () => {
      const objectMargin = margin({ margin: { top: "10px" } });
      expect(objectMargin.join("")).toEqual(expect.stringContaining('margin-top: 10px;'));
    });

    it('should return a string containing the two declared margins in the object', () => {
      const multipleObjectMargin = margin({ margin: { top: "10px", left: "15px" } });

      expect(multipleObjectMargin.join("")).toEqual(expect.stringContaining('margin-top: 10px;'));
      expect(multipleObjectMargin.join("")).toEqual(expect.stringContaining('margin-left: 15px;'));
    });
  });

  describe("Test the padding styled-components helper", () => {
    it('should return nothing if no padding is passed', () => {
      const noPadding = padding({});
      expect(noPadding).toBe('');
    });

    it('should return a string containing the declared string padding', () => {
      const stringPadding = padding({ padding: "10px" });
      expect(stringPadding.join("")).toEqual(expect.stringContaining('padding: 10px;'));
    });

    it('should return a string containing the declared object padding', () => {
      const objectPadding = padding({ padding: { top: "10px" } });
      expect(objectPadding.join("")).toEqual(expect.stringContaining('padding-top: 10px;'));
    });

    it('should return a string containing the two declared paddings in the object', () => {
      const multipleObjectPadding = padding({ padding: { top: "10px", left: "15px" } });

      expect(multipleObjectPadding.join("")).toEqual(expect.stringContaining('padding-top: 10px;'));
      expect(multipleObjectPadding.join("")).toEqual(expect.stringContaining('padding-left: 15px;'));
    });
  });

  describe("Test the border styled-components helper", () => {
    it('should return nothing if no border is passed', () => {
      const noBorder = border({});
      expect(noBorder).toBe('');
    });

    it('should return a string containing the declared string border', () => {
      const stringBorder = border({ border: "10px" });
      expect(stringBorder.join("")).toEqual(expect.stringContaining('border: 10px;'));
    });

    it('should return a string containing the declared object border', () => {
      const objectBorder = border({ border: { top: "10px" } });
      expect(objectBorder.join("")).toEqual(expect.stringContaining('border-top: 10px;'));
    });

    it('should return a string containing the two declared borders in the object', () => {
      const multipleObjectBorder = border({ border: { top: "10px", left: "15px" } });

      expect(multipleObjectBorder.join("")).toEqual(expect.stringContaining('border-top: 10px;'));
      expect(multipleObjectBorder.join("")).toEqual(expect.stringContaining('border-left: 15px;'));
    });
  });


});