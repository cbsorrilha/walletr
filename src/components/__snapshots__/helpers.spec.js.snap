// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component helpers Test  the media helper should match the snapshots 1`] = `
Object {
  "desktop": 992,
  "mobile": 576,
  "tablet": 768,
}
`;
